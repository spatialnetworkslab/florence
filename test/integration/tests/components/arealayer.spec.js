/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/arealayer')
  })

  describe('AreaLayer Tests', () => {
    it('[render] area layer rendered', () => {
      cy.get('.area').should('have.length', 2)
    })

    it('[interaction] hovering over area layer changes color', () => {
      const area = cy.get('.area').first()
      area.should('have.attr', 'fill', '#af8dc3')
      area.trigger('mousemove')
      area.should('have.attr', 'fill', '#eee0cb')
    })
  })
})
