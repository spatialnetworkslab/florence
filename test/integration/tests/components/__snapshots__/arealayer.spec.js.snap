exports[`Assertions > AreaLayer Tests > [render] area layer snapshot correctly #0`] = `
<svg width="700" height="500">
  <defs>
    <clipPath id="clip-sc0">
      <rect x="0" y="0" width="700" height="500"></rect>
    </clipPath>
    <clipPath id="clip-sc0-data">
      <rect x="30" y="20" width="650" height="450" fill="white"></rect>
    </clipPath>
  </defs>
  <g class="section" clip-path="url(#clip-sc0)">
    <g class="area-layer">
      <path
        class="area"
        d="M62.5,416L95,416L127.5,402.5L160,402.5L192.5,393.5L225,398L257.5,407L290,389L322.5,380L355,402.5L387.5,389L420,380L452.5,380L485,375.5L517.5,402.5L550,375.5L582.5,384.5L615,362L647.5,398L680,371L680,470L647.5,470L615,470L582.5,470L550,470L517.5,470L485,470L452.5,470L420,470L387.5,470L355,470L322.5,470L290,470L257.5,470L225,470L192.5,470L160,470L127.5,470L95,470L62.5,470Z"
        fill="#af8dc3"
        stroke="none"
        stroke-width="0"
        opacity="1"
      ></path>
      <path
        class="area"
        d="M62.5,348.5L95,362L127.5,326L160,326L192.5,339.5L225,330.5L257.5,353L290,335L322.5,290L355,276.5L387.5,308L420,308L452.5,326L485,312.5L517.5,312.5L550,285.5L582.5,303.5L615,317L647.5,308L680,317L680,371L647.5,398L615,362L582.5,384.5L550,375.5L517.5,402.5L485,375.5L452.5,380L420,380L387.5,389L355,402.5L322.5,380L290,389L257.5,407L225,398L192.5,393.5L160,402.5L127.5,402.5L95,416L62.5,416Z"
        fill="#7fbf7b"
        stroke="none"
        stroke-width="0"
        opacity="1"
      ></path>
    </g>
    <g class="x-axis">
      <g class="line-layer">
        <path
          class="line"
          d="M30,470.45L30,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M95,470.45L95,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M160,470.45L160,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M225,470.45L225,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M290,470.45L290,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M355,470.45L355,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M420,470.45L420,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M485,470.45L485,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M550,470.45L550,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M615,470.45L615,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M680,470.45L680,465.05"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
      </g>
      <g class="label-layer">
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="30"
          opacity="1"
          transform="
          rotate(0, 
          30, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          0
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="95"
          opacity="1"
          transform="
          rotate(0, 
          95, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          2
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="160"
          opacity="1"
          transform="
          rotate(0, 
          160, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          4
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="225"
          opacity="1"
          transform="
          rotate(0, 
          225, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          6
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="290"
          opacity="1"
          transform="
          rotate(0, 
          290, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          8
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="355"
          opacity="1"
          transform="
          rotate(0, 
          355, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          10
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="420"
          opacity="1"
          transform="
          rotate(0, 
          420, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          12
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="484.99999999999994"
          opacity="1"
          transform="
          rotate(0, 
          484.99999999999994, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          14
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="550"
          opacity="1"
          transform="
          rotate(0, 
          550, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          16
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="615"
          opacity="1"
          transform="
          rotate(0, 
          615, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          18
        </text>
        <text
          class="label"
          y="462.8"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="680"
          opacity="1"
          transform="
          rotate(0, 
          680, 
          462.8)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          20
        </text>
      </g>
    </g>
    <g class="y-axis">
      <g class="line-layer">
        <path
          class="line"
          d="M29.54,470L23.96,470"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,425L23.96,425"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,380L23.96,380"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,335L23.96,335"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,290L23.96,290"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,245L23.96,245"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,200L23.96,200"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,155L23.96,155"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,110L23.96,110"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,65L23.96,65"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M29.54,20L23.96,20"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
      </g>
      <g class="label-layer">
        <text
          class="label"
          y="470"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          470)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          0
        </text>
        <text
          class="label"
          y="425"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          425)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          10
        </text>
        <text
          class="label"
          y="380"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          380)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          20
        </text>
        <text
          class="label"
          y="335"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          335)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          30
        </text>
        <text
          class="label"
          y="290"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          290)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          40
        </text>
        <text
          class="label"
          y="245"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          245)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          50
        </text>
        <text
          class="label"
          y="200"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          200)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          60
        </text>
        <text
          class="label"
          y="155.00000000000003"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          155.00000000000003)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          70
        </text>
        <text
          class="label"
          y="109.99999999999999"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          109.99999999999999)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          80
        </text>
        <text
          class="label"
          y="65"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          65)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          90
        </text>
        <text
          class="label"
          y="20"
          fill="black"
          stroke="none"
          stroke-width="0"
          x="22.57142857142857"
          opacity="1"
          transform="
          rotate(0, 
          22.57142857142857, 
          20)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="end"
          dominant-baseline="middle"
        >
          100
        </text>
      </g>
    </g>
  </g>
</svg>
`;
